tests/conftest.py

```diff
@@ -67,17 +57,37 @@ def mock_product_test_spotlight(
         constraints=MyProductConstraints,
         opportunity_properties=MyOpportunityProperties,
         order_parameters=MyOrderParameters,
-        backend=product_backend,
+        create_order=mock_create_order,
+        search_opportunities=mock_search_opportunities,
     )
 
 
 @pytest.fixture
 def stapi_client(
-    root_backend, mock_product_test_spotlight, base_url: str
+    mock_product_test_spotlight,
+    base_url: str,
+    orders_db: InMemoryOrderDB | None = None,
+    opportunities: list[Opportunity] | None = None,
 ) -> Iterator[TestClient]:
-    root_router = RootRouter(root_backend)
+    @asynccontextmanager
+    async def lifespan(app: FastAPI) -> AsyncIterator[dict[str, Any]]:
+        try:
+            yield {
+                "_orders_db": orders_db if orders_db is not None else InMemoryOrderDB(),
+                "_opportunities": opportunities
+                if opportunities is not None
+                else mock_test_spotlight_opportunities(),
```

Comment:
Allow me to clarify a later point I made on my previous comment. I'd rename `mock_test_spotlight_opportunities` because it is long, redundant, and overly specific to something like `mock_opportunities`. Then if you make this parameter named the same that fixture will be used by default and you can simplify here. (I also removed the conditional here for the db per my above suggestion).

```
    mock_opportunities: list[Opportunity],
) -> Iterator[TestClient]:
    root_router = RootRouter(root_backend)
    @asynccontextmanager
    async def lifespan(app: FastAPI) -> AsyncIterator[dict[str, Any]]:
        try:
            yield {
                "_orders_db": InMemoryOrderDB(),
                "_opportunities": mock_opportunities,
```